sql/core/src/test/scala/org/apache/spark/sql/test/SharedSparkSessionWithFreshCopy.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.test
+
+import org.scalatest.Suite
+
+// A variant of SharedSparkSession. It creates a fresh copy of the spark session before each test.
+trait SharedSparkSessionWithFreshCopy extends SharedSparkSession { self: Suite =>
```

Comment:
If `SharedSparkSession` means sharing spark session across tests, `SharedSparkSessionWithFreshCopy` seems a weird name, because it doesn't share session between tests.

Reply:
this is a good point. Do you have any suggestions? maybe `SharedSparkContextWithFreshSparkSession`?